### PR TITLE
Prefer stateless function in platform

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

This refactors code in the `src/platform` directory to conform to the [prefer-stateless-function](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md) ESLint rule that we currently have disabled.

Related to #12224

## Testing done

Local. These are my uncommitted changes for testing:

```diff
diff --git a/.eslintrc b/.eslintrc
index 25ad07878..4a15e8d56 100644
--- a/.eslintrc
+++ b/.eslintrc
@@ -58,7 +58,6 @@
 
     /* || react plugin || */
     "react/no-multi-comp": 0, // Leave organization to code reviewer discretion.
-    "react/prefer-stateless-function": 0, // Leave statelessness to code reviewer discretion.
     "react/jsx-key": 2,
     "react/jsx-pascal-case": 2,
     "react/jsx-no-duplicate-props": 2,
diff --git a/package.json b/package.json
index 16f0001a8..643fc644b 100644
--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
     "heroku-postbuild": "sh script/heroku-postbuild.sh",
     "lint": "npm run lint:js && npm run lint:sass",
     "lint:changed": "npm run lint:js:changed && npm run lint:sass",
-    "lint:js": "eslint --quiet --ext .js --ext .jsx .",
+    "lint:js": "eslint --quiet --ext .js --ext .jsx src/platform",
     "lint:circle": "eslint --quiet --ext .js --ext .jsx -c circle.eslint.json .",
     "lint:js:fix": "eslint --fix --quiet --ext .js --ext .jsx .",
     "lint:js:changed": "LIST=`git diff-index --name-only --cached HEAD | grep \"\\.js[x]\\{0,1\\}$\"`; if [ \"$LIST\" ]; then eslint --quiet $LIST; fi",
```

## Screenshots

With the local testing changes in place on master:

image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms/components/FormFooter.jsx [hunk 1]
```diff
@@ -1,32 +1,26 @@
...
+export default function FormFooter({ formConfig, currentLocation }) {
+  const GetFormHelp = formConfig.getHelp;
 
-    if (
-      currentLocation &&
-      currentLocation.pathname.replace(/\/$/, '').endsWith('confirmation')
-    ) {
-      return null;
-    }
+  if (currentLocation?.pathname.replace(/\/$/, '').endsWith('confirmation')) {
```
Comment (bkjohnson) [inline src/platform/forms/components/FormFooter.jsx:6]: I added in optional chaining here.

File: src/platform/forms/save-in-progress/SaveStatus.jsx [hunk 2]
```diff
@@ -5,68 +5,67 @@ import moment from 'moment';
...
-    if (form.lastSavedDate) {
-      const savedAt = moment(form.lastSavedDate);
-      savedAtMessage = ` Last saved at ${savedAt.format(
-        'MMM D, YYYY [at] h:mm a',
-      )}`;
-    } else {
-      savedAtMessage = '';
-    }
-
-    const savedStatus = form.autoSavedStatus;
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveStatus.jsx:22]: This const has been replaced with `autoSavedStatus` from the form object.

File: src/platform/forms/save-in-progress/SaveStatus.jsx [hunk 3]
```diff
@@ -5,68 +5,67 @@ import moment from 'moment';
...
-
-    const hasError =
-      saveErrors.has(savedStatus) &&
-      ((savedStatus === SAVE_STATUSES.noAuth && !isLoggedIn) ||
-        savedStatus !== SAVE_STATUSES.noAuth);
+function SaveStatus({
+  form: { lastSavedDate, autoSavedStatus },
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveStatus.jsx:9]: Destructuring to get `autoSavedStatus`

File: src/platform/forms/save-in-progress/SaveStatus.jsx [hunk 4]
```diff
@@ -5,68 +5,67 @@ import moment from 'moment';
...
-    const signInLink = (
-      <SignInLink
-        className="va-button-link"
-        isLoggedIn={isLoggedIn}
-        showLoginModal={showLoginModal}
-        toggleLoginModal={toggleLoginModal}
-      >
-        Sign in to save your form in progress
-      </SignInLink>
-    );
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveStatus.jsx:38]: This was removed in favor of inlining the jsx

File: src/platform/forms/components/FormFooter.jsx (hunk 1 above)
Comment (jhonnyoddball) [reply src/platform/forms/components/FormFooter.jsx:6]: This is great!

**Review by jhonny218 — APPROVED** [review]
LGTM

File: src/platform/user/profile/vet360/components/base/Vet360Transaction.jsx [hunk 5]
```diff
@@ -10,51 +10,51 @@ import {
...
-export default class Vet360Transaction extends React.Component {
-  static propTypes = {
-    children: PropTypes.node.isRequired,
-    refreshTransaction: PropTypes.func.isRequired,
-    title: PropTypes.string.isRequired,
-    transaction: PropTypes.object,
-  };
+function Vet360Transaction(props) {
+  const {
+    id,
```
Comment (erikphansen) [inline src/platform/user/profile/vet360/components/base/Vet360Transaction.jsx:15]: thank you for not destructuring this in the function argument :-D

**Review by erikphansen — APPROVED** [review]

**Pull request merged**